### DISCUSSION TITLE
FluxCD limits

### DISCUSSION THREAD

**Discussion by Moon1706** [fluxcd/flux2#4291] [Q&A]
Hey folks! 
Could anyone write limits for FluxCD, please? Our team wanted to migrate from ArgoCD to FluxCD because ArgoCD really badly works with more than 5k applications (problems with sync and performance). Is it the same in FluxCD?

**Comment by kingdonb** [comment]
I know I have heard of a Flux user recently with 5k applications (this was a conversation in the Slack channel, if you read the past week or two of public posts in #flux I'm sure you will find it) who was given advice about how to resolve their issue so the performance would be acceptable and they could scale past this limit. There are guides for scaling Flux that will show how to add resources ([vertical scaling](https://fluxcd.io/flux/installation/configuration/vertical-scaling/)) and how to add more instances of the controllers ([horizontal scaling](https://fluxcd.io/flux/installation/configuration/sharding/)) so that you can scale to tens of thousands of releases.

There are also some paths in Flux that, in spite of recent optimizations to perform better than before, still don't work as well as we'd like (case in point: Helm Legacy Helm Repositories), and there are better paths superseding those that have improved scalability ([Helm with OCI repository source](https://fluxcd.io/flux/cheatsheets/oci-artifacts/#helm-oci), and the [standalone `OCIRepository`](https://fluxcd.io/flux/cheatsheets/oci-artifacts/) Flux source kind).

There are many optimizations in the latest versions of Flux, so that you can handle the limits of scale. We don't publish limits for these concerns, (number one because we have limited resources and testing the limits of scale costs a lot of money!) Number two because we do always take feedback and work rapidly with our users/customers to ensure that you can pass the limits when you need to.

The design of Flux includes many optimizations that are always improving. If you find that your enterprise is struggling to scale past 5k applications, we can definitely help. No, it's not the same, Flux is designed from the ground for scalability concerns and has adapted to meet a diverse set of needs. Hope this answer helps you out!
